server/src/main/java/org/opensearch/extensions/ExtensionDependencyResponse.java

```diff
@@ -0,0 +1,70 @@
+/*
+* Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.transport.TransportResponse;
+
+/**
+ * The response for getting the Extension Dependency.
+ *
+ * @opensearch.internal
+ */
+public class ExtensionDependencyResponse extends TransportResponse {
+    private List<DiscoveryExtensionNode> dependency;
+
+    public ExtensionDependencyResponse(List<DiscoveryExtensionNode> dependency) {
+        this.dependency = dependency;
+    }
+
+    public ExtensionDependencyResponse(StreamInput in) throws IOException {
+        super(in);
+        int size = in.readVInt();
+        dependency = new ArrayList<>(size);
+        for (int i = 0; i < size; i++) {
+            dependency.add(new DiscoveryExtensionNode(in));
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
```

Comment:
Can we have `super.writeTo(out);` here?

Reply:
yes, I will add that in next commit

Reply:
We probably want to remove the `super(in)` on the stream constructor, then.  I think it's a no-op for responses anyway, but I think the two methods should match up.